### PR TITLE
Adds wdl that tests joint VCF filtering tools

### PR DESCRIPTION
**Pull request by meganshand**
This adds a small test case for the WDL of the filtering pipeline. This still has indels and snps separated out. I can combine them if needed, but we'd like to use different annotations for each mode. This also doesn't actually apply the final filtering (with a threshold) since we still need to add a step to determine the correct threshold. The final VCFs from this workflow should have SCORE INFO annotations for each site.

This takes in an array of VCFs (and outputs an array of VCFs) because this is an option for large callsets in the WARP joint genotyping WDL which is where this WDL will eventually be integrated.

This test only ensures that the WDL runs and doesn't compare to expected results (the same as the other WDL tests in this repo).

### CODE REVIEW

File: scripts/filtering_wdl/jsons/filtering_travis.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  "JointVcfFiltering.vcf": ["/home/runner/work/gatk/gatk/src/test/resources/large/filteringJointVcf/test_10_samples.22.avg.vcf.gz",
+    "/home/runner/work/gatk/gatk/src/test/resources/large/filteringJointVcf/test_10_samples.23.avg.vcf.gz"],
+  "JointVcfFiltering.vcf_index": ["/home/runner/work/gatk/gatk/src/test/resources/large/filteringJointVcf/test_10_samples.22.avg.vcf.gz.tbi",
+    "/home/runner/work/gatk/gatk/src/test/resources/large/filteringJointVcf/test_10_samples.23.avg.vcf.gz.tbi"],
+  "JointVcfFiltering.sites_only_vcf": "/home/runner/work/gatk/gatk/src/test/resources/large/filteringJointVcf/test_10_samples.sites_only.vcf.gz",
+  "JointVcfFiltering.sites_only_vcf_index": "/home/runner/work/gatk/gatk/src/test/resources/large/filteringJointVcf/test_10_samples.sites_only.vcf.gz.tbi",
+  "JointVcfFiltering.basename": "test_10_samples",
+  "JointVcfFiltering.indel_sensitivity_threshold": 97.0,
+  "JointVcfFiltering.snp_sensitivity_threshold": 99.9,
+  "JointVcfFiltering.snp_annotations": "-A ReadPosRankSum -A FS -A SOR -A QD -A AVERAGE_TREE_SCORE -A AVERAGE_ASSEMBLED_HAPS -A AVERAGE_FILTERED_HAPS",
```
Comment (samuelklee) [inline scripts/filtering_wdl/jsons/filtering_travis.json:12]: Following up on comments elsewhere: would it be possible to use the `AVERAGE_ASSEMBLED_HAPS` and `AVERAGE_FILTERED_HAPS` annotations for INDELs, eventually?

File: scripts/filtering_wdl/JointVcfFiltering.wdl [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+		gatk --java-options "-Xmx${command_mem}m" \
+		ScoreVariantAnnotations \
+		~{interval_arg} \
+		-V ~{vcf} \
+		-O ~{basename}.~{mode} \
+		--python-script ~{scoring_python_script} \
+		--model-prefix ~{basename} \
+		~{annotations} \
+		-mode ~{mode} \
+		--resource:extracted-training,training=true,calibration=false ~{extracted_training_vcf} \
```
[the remark this answers is not in the document]
Comment (meganshand) [reply scripts/filtering_wdl/JointVcfFiltering.wdl:239]: Ah ok, this wasn't super clear to me. I changed it to extracted as you suggested above.

File: scripts/filtering_wdl/JointVcfFiltering.wdl [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+		-O ~{basename} \
+		--python-script ~{python_script} \
+		--hyperparameters-json ~{hyperparameters_json} \
+		-mode ~{mode_uc}
+
+	}
+	output {
+		File scorer = "~{basename}.~{mode_lc}.scorer.pkl"
+		File training_scores = "~{basename}.~{mode_lc}.trainingScores.hdf5"
+		File truth_scores = "~{basename}.~{mode_lc}.calibrationScores.hdf5"
```
[the remark this answers is not in the document]
Comment (samuelklee) [reply scripts/filtering_wdl/JointVcfFiltering.wdl:191]: The *.annot.hdf5 output of the extract tool, which contains the (positive) training and calibration data, remains unchanged. However, an additional *.unlabeled.annot.hdf5 output is generated, which contains the unlabeled data separately.

File: scripts/vcf_site_level_filtering_wdl/JointVcfFiltering.wdl [hunk 4]
```diff
@@ -160,6 +160,7 @@ task ExtractVariantAnnotations {
 		File annots = "~{basename}.~{mode}.annot.hdf5"
 		File extracted_training_vcf = "~{basename}.~{mode}.vcf.gz"
 		File extracted_training_vcf_index = "~{basename}.~{mode}.vcf.gz.tbi"
+		Array[File] outputs = glob("~{basename}.~{mode}.*")
```
[the remark this answers is not in the document]
Comment (meganshand) [reply scripts/vcf_site_level_filtering_wdl/JointVcfFiltering.wdl:163]: Yeah, that was my intention (to both glob and have them listed out as individual files). This was to make it easier to import the individual files to other tasks and use them in the command line appropriately.

**Comment by meganshand** [comment]
@samuelklee would you like to merge this branch into yours? Or did you want to PR your branch first?

**Comment by samuelklee** [comment]
Hmm, I guess I can go ahead and merge into my branch.

File: scripts/vcf_site_level_filtering_wdl/JointVcfFiltering.wdl (hunk 4 above)
[the remark this answers is not in the document]
Comment (meganshand) [reply scripts/vcf_site_level_filtering_wdl/JointVcfFiltering.wdl:163]: Gotcha, makes sense!

**Pull request merged**